tests/scripts/generate_test_driver.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+
+# generate_test_driver.py
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+"""
+Generate a TF-PSA-Crypto test driver
+"""
+import itertools
+import sys
+
+from fnmatch import fnmatch
+from pathlib import Path
+from typing import Iterable, List, Set
+
+import scripts_path # pylint: disable=unused-import
+from mbedtls_framework import build_tree
+from mbedtls_framework import test_driver
+
+from config import TFPSACryptoConfig
+
+EXCLUDE_FILES = {
+    "asn1*",
+    "base64*",
+    "crypto_builtin_key_derivation.h",
+    "lmots*",
+    "lms.c",
+    "md.c",
+    "memory_buffer_alloc.c",
+    "nist_kw.c",
+    "pem.c",
+    "pk*",
+    "platform*",
+    "threading*",
+}
+
+IDENTIFIER_PREFIXES = {
+    "MBEDTLS_",
+    "PSA_",
+    "TF_PSA_CRYPTO_",
+    "mbedtls_",
+    "psa_",
+    "tf_psa_crypto_",
+}
+
+def iter_code_files(root: Path) -> Iterable[Path]:
+    """
+    Iterate over all "*.c" and "*.h" files found recursively under the `include`
+    and `src` subdirectories of `root`.
+    """
+    for directory in ("include", "src"):
+        directory_path = root / directory
+        for ext in (".c", ".h"):
+            yield from directory_path.rglob(f"*{ext}")
+
+def get_src_relpaths(builtin: Path, exclude_files: Set[str]) -> List[Path]:
+    """
+    Return the relative paths of all *.c and *.h files under `builtin`,
+    excluding those whose names match any of the patterns in `exclude_files`.
```

Comment:
“file name” either refers to the base name or to the full path (not necessarily absolute), depending on who you ask. Please clarify.


```suggestion
    excluding those whose basenames match any of the patterns in `exclude_files`.
```